src/pg/index.js

```diff
@@ -35,65 +21,18 @@ export default (opts = {}) =>
       ...opts,
     };
 
-    const watchers = new Set();
-    let timestamp = Date.now();
-
-    function getArgKey() {
-      /* TODO */
-    }
-
-    async function poll() {
-      if (!watchers.size) return;
-      const result = await dbPoll(timestamp, pgOptions);
-
-      for (const { query, push } of watchers) {
-        for (const [object] of result) {
-          const payload = [];
-
-          for (const node of query) {
-            const args = decodeArgs(node);
-            if (isPlainObject(args)) {
-              if (filterObject(args, object))
-                payload.push({
-                  ...object,
-                  $ref: object.$key,
-                  $key: getArgKey(args, object),
-                });
-            } else {
-              if (object.id === node.key) payload.push(object);
-            }
-          }
-
-          push(wrap(slice(encodeGraph(payload), query).known, prefix));
-        }
-      }
-    }
-
-    setInterval(poll, pgOptions.pollInterval);
+    const myTransaction = new Transaction({ client, pgOptions });
```

Comment:
First on naming: For me, this object is not a transaction, it is just a wrapper around the PG client. A transaction is what you do with SQL `BEGIN`, `COMMIT` and `ROLLBACK` statements. We will have to add support for transactions soon, so it is better to name this something else. I suggest naming it DbWrapper, so we do `const db = new DbWrapper(...)`, and you can rename the methods so it becomes `db.read()` rather than `db.dbRead()`.

Second, we should only accept a pool object (not client). This is required for both stability (clients disconnect eventually and don't reconnect automatically; we need this to last the lifetime of a server process) and performance (clients can only do one query at a time, no parallelism). I suggest we replace the `client` option here with `pgPool`.

Finally, just to make sure we are on the same page (no change required): this library will eventually need to get PG connections in 4 different ways. 1, 2 and 3 are based on the behavior of other PG libraries in NPM, such as migrations and graphile-worker; 4 is specific to us.

1. User does not provide anything. Library should create a Pool using PSQL environment variables. This is helpful when we need to write one-off scripts.
2. User provides the credentials at the time of server initialization (in the `.use(pg(...))` call), and the library uses these to create a Pool. Also helpful for one-off scripts.
3. User provides a [pool](https://node-postgres.com/api/pool) object at the time of server initialization (in the `.use(pg(...))` call). This is what we are supporting here.
4. We get a client in each query (in the `.read(query, options)` or `.write(change, options)` calls). We need this in Lego for proper transactions (atomicity) in Lego writes. We will implement it using a separate middleware, called `transaction` which might add an option like `graffyPgDbWrapper` to the options object. We can build this in another PR, this is already quite big.

Reply:
yeah, the handling of transaction will be in the next PR